compiler/noirc_evaluator/src/ssa/ir/instruction.rs

```diff
@@ -386,15 +387,18 @@ impl Instruction {
             | IfElse { .. }
             | ArraySet { .. } => true,
 
+            // Store instructions must be removed by DIE in acir code, any load
+            // instructions should already be unused by that point.
+            Store { .. } => matches!(function.runtime(), RuntimeType::Acir(_)),
```

Comment:
With the change to mem2reg, stores weren't being removed otherwise for some reason (even though the comment claims to just remove certain unneeded stores). So I had to change DIE to remove all stores here.